src/components/ha-hls-player.ts

```diff
@@ -126,13 +125,24 @@ class HaHLSPlayer extends LitElement {
     }
 
     this._useExoPlayer = await useExoPlayerPromise;
-    let hevcRegexp: RegExp;
-    let masterPlaylist: string;
-    if (this._useExoPlayer) {
-      hevcRegexp = /CODECS=".*?((hev1)|(hvc1))\..*?"/;
-      masterPlaylist = await (await masterPlaylistPromise).text();
+    const masterPlaylist = await (await masterPlaylistPromise).text();
+
+    // Parse playlist assuming it is a master playlist. Match group 1 is whether hevc, match group 2 is regular playlist url
+    // See https://tools.ietf.org/html/rfc8216 for HLS spec details
+    const playlistRegexp = /#EXT-X-STREAM-INF:.*?(?:CODECS=".*?(hev1|hvc1)?\..*?".*?)?(?:\n|\r\n)(.+)/g;
+    const match = playlistRegexp.exec(masterPlaylist);
+
+    // Get the regular playlist url from the input (master) playlist, falling back to the input playlist if necessary
+    // This avoids the player having to load and parse the master playlist again before loading the regular playlist
+    let playlist_url: string;
+    if (match !== null && playlistRegexp.exec(masterPlaylist) === null) {
```

Comment:
It still is not clear what we are doing here, why do we do the regex a second time and why should it not match?

Reply:
The second regex is to test whether there's another #EXT-X-STREAM-INF line which would mean that our master playlist has more than one stream. We should actually not encounter this case currently as we are currently only include one stream in our master playlist, but in case we change things, this safely sends the master_playlist instead of skipping to the regular playlist when there is more than one stream available.

Reply:
Please add comment for that, can we also give the various match part logical variable names? like: 
`const isHevc = match && match[1];`

Reply:
I pushed a change, does that look better? I haven't had a chance to test it yet

Reply:
Yes, that looks better. Let me know when you tested it :-)

Reply:
Just tested it and seems to be working fine